nearlib/test/integration.test.js

```diff
@@ -167,6 +167,81 @@ test('create account with a new key and then view account returns the created ac
     expect(aliceAccountAfterCreation.amount).toBe(aliceAccountBeforeCreation.amount - amount);
 });
 
+describe('with access key', function () {
+    let oldLog;
+    let logs;
+    let contractId = 'test_contract_' + Date.now();
+    let newAccountId;
+    let newAccountKeyPair;
+
+    beforeAll(async () => {
+        const keyWithRandomSeed = await KeyPair.fromRandomSeed();
+        const createAccountResponse = await account.createAccount(
+            contractId,
+            keyWithRandomSeed.getPublicKey(),
+            10,
+            aliceAccountName);
+        await nearjs.waitForTransactionResult(createAccountResponse);
+        await keyStore.setKey(contractId, keyWithRandomSeed);
+        const data = [...fs.readFileSync('../tests/hello.wasm')];
+        await nearjs.waitForTransactionResult(
+            await nearjs.deployContract(contractId, data));
+
+    });
+
+    beforeEach(async () => {
+        oldLog = console.log;
+        logs = [];
+        console.log = function() {
+            logs.push(Array.from(arguments).join(' '));
+        };
+
+        newAccountId = await generateUniqueString('create.account.test');
+        newAccountKeyPair = await KeyPair.fromRandomSeed();
+        const createAccountResponse = await account.createAccount(
+            newAccountId,
+            newAccountKeyPair.getPublicKey(),
+            0,
+            aliceAccountName);
+        await nearjs.waitForTransactionResult(createAccountResponse);
+        await keyStore.setKey(newAccountId, newAccountKeyPair);
+    });
+
+    afterEach(async () => {
+        console.log = oldLog;
+    });
+
+    test('make function calls using access key', async () => {
+        // Adding access key
+        const keyForAccessKey = await KeyPair.fromRandomSeed();
+        const addAccessKeyResponse = await account.addAccessKey(
+            newAccountId,
+            keyForAccessKey.getPublicKey(),
+            contractId,
+            '',
+            '',
+            0,
+        );
+        await nearjs.waitForTransactionResult(addAccessKeyResponse);
+        // Replacing public key for the account with the access key
+        await keyStore.setKey(newAccountId, keyForAccessKey);
+        // test that load contract works and we can make calls afterwards
+        const contract = await nearjs.loadContract(contractId, {
+            viewMethods: ['hello', 'getValue'],
+            changeMethods: ['setValue'],
+            sender: newAccountId,
+        });
+        const helloResult = await contract.hello({ name: 'trex' });
```

Comment:
looks like `contract.hello` call isn't necessary for test

also I think at this point it makes sense to extract "test that we can write value and read back" into a helper function (cause it's useful for multiple tests)

Reply:
Yeah. I guess view calls are not needed, unless we change a logic.